lib/everypolitician/popolo/collection.rb

```diff
@@ -18,6 +18,18 @@ def -(other)
         documents.reject { |d| other_ids.include?(d.id) }
       end
 
+      def find_by(attributes = {})
+        where(attributes).first
+      end
+
+      def where(attributes = {})
+        find_all do |object|
+          !attributes.collect do |k,v|
+            object.send(k) == v
+          end.include?(false)
```

Comment:
Waaay nicer! :ok_hand: I *knew* there had to be a tidier way to do this but we couldn't work it on late on a Friday :smile: